### PR TITLE
feat: Support watching multiple namespaces

### PR DESCRIPTION
**Pull request by ciarams87**
### Proposed changes
Currently, the NGINX Ingress Controller can watch either all namespaces or a single namespace. This commit adds the ability to watch multiple namespaces (defined using by passing a comma separated list of namespaces to the `-watch-namespace` CLI argument instead of a single name). This is achieved by creating an informer per namespace and kind we observe.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by shaun-nx — APPROVED** [review]
👍

**Review by vepatel — APPROVED** [review]

File: cmd/nginx-ingress/flags.go [hunk 1]
```diff
@@ -331,11 +340,31 @@ func validationChecks() {
 	}
 }
 
+// validateNamespaceNames validates the namespaces are in the correct format
```
Comment (vepatel) [inline cmd/nginx-ingress/flags.go:343]: Can we get a unit test for this if possible? for cases like:
`-watch-namespace=watched_namespace,`
`-watch-namespace="watched_namespace1,watched_namespace2"`
`-watch-namespace="watched_namespace1","watched_namespace2"`

It think there are regex checks in place for invalid chars already

File: docs/content/installation/installation-with-helm.md [hunk 2]
```diff
@@ -183,7 +183,7 @@ The following tables lists the configurable parameters of the NGINX Ingress Cont
 |``controller.replicaCount`` | The number of replicas of the Ingress Controller deployment. | 1 |
 |``controller.ingressClass`` | A class of the Ingress Controller. An IngressClass resource with the name equal to the class must be deployed. Otherwise, the Ingress Controller will fail to start. The Ingress Controller only processes resources that belong to its class - i.e. have the "ingressClassName" field resource equal to the class. The Ingress Controller processes all the VirtualServer/VirtualServerRoute/TransportServer resources that do not have the "ingressClassName" field for all versions of kubernetes. | nginx |
 |``controller.setAsDefaultIngress`` | New Ingresses without an ingressClassName field specified will be assigned the class specified in `controller.ingressClass`. | false |
-|``controller.watchNamespace`` | Namespace to watch for Ingress resources. By default the Ingress Controller watches all namespaces. | "" |
+|``controller.watchNamespace`` | Comma separated list of namespaces to watch for Ingress resources. By default the Ingress Controller watches all namespaces. | "" |
```
Comment (vepatel) [inline docs/content/installation/installation-with-helm.md:186]: should we remove `Ingress resources` from here completely as there could by any resource in those ns?

**Review by vepatel — APPROVED** [review]
👍🏼

**Pull request merged**